generator/rangewriter.go

```diff
@@ -17,13 +18,21 @@ type RangeWriter struct {
 	Current   parser.Position
 	inLiteral bool
 	w         io.Writer
+
+	// Extract strings
+	extractStrings bool
+	strings        []string
 }
 
 func (rw *RangeWriter) closeLiteral(indent int) (r parser.Range, err error) {
 	rw.inLiteral = false
-	_, err = rw.write("\")\n")
-	if err != nil {
-		return
+	if rw.extractStrings {
```

Comment:
I don't understand what's going on here. Probably because I haven't read the rest yet, but maybe a comment would help.

Reply:
Cleaned this up